### PR TITLE
Fix CondVar::TimedWait for Windows

### PR DESCRIPTION
**Pull request by eniac1024**
Based on https://github.com/microsoft/STL/issues/369 
They fixed the issue in `std::condition_variable_any` but not in `std::condition_variable`, which is currently used in rocksdb repo. So we need to implement the work around regardless of `_MSVC_STL_UPDATE`.

### CODE REVIEW

File: port/win/port_win.cc [hunk 1]
```diff
@@ -101,8 +101,12 @@ bool CondVar::TimedWait(uint64_t abs_time_us) {
   std::unique_lock<std::mutex> lk(mu_->getLock(), std::adopt_lock);
 
   // Work around https://github.com/microsoft/STL/issues/369
-#if defined(_MSC_VER) && \
-    (!defined(_MSVC_STL_UPDATE) || _MSVC_STL_UPDATE < 202008L)
+  // std::condition_variable_any::wait_for had a fix,
+  // use the following two lines of directives instead
+  // #if defined(_MSC_VER) && \
+  //     (!defined(_MSVC_STL_UPDATE) || _MSVC_STL_UPDATE < 202008L)
```
Comment (ajkr) [inline port/win/port_win.cc:107]: Sorry I don't know when someone should use these two lines of directives. If they're not needed can we delete them?

**Review by ajkr — CHANGES_REQUESTED** [review]
What does "for vs2019" mean in the title? The description and code appear to generalize the workaround to non-vs2019.

File: port/win/port_win.cc (hunk 1 above)
Comment (eniac1024) [reply port/win/port_win.cc]: Removed

**Comment by eniac1024** [comment]
> What does "for vs2019" mean in the title? The description and code appear to generalize the workaround to non-vs2019.
You're right, it's a generalized workaround, and the title is confusing. This issue started from compiling with VS2019 which bumped the STL version, and it was all good with previous VS version.

**Review by ajkr — APPROVED** [review]
LGTM, thanks!

**Comment by cbi42** [comment]
CI complains about code format. Need to run `make format`.

**Comment by ajkr** [comment]
> CI complains about code format. Need to run `make format`.

Done

**Pull request closed without merging**